pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -355,8 +355,9 @@ void setIdealState(String realtimeTableName, IdealState idealState) {
    * This method moves the segment file from another location to its permanent location.
    * When splitCommit is enabled, segment file is uploaded to the segmentLocation in the committingSegmentDescriptor,
    * and we need to move the segment file to its permanent location before committing the segment metadata.
+   * Return the permanent location of the segment if the commit succeeds.
    */
-  public void commitSegmentFile(String realtimeTableName, CommittingSegmentDescriptor committingSegmentDescriptor)
+  public String commitSegmentFile(String realtimeTableName, CommittingSegmentDescriptor committingSegmentDescriptor)
```

Comment:
check the descriptor for peer  scheme in this method, and move only if it is not peer scheme. leave the descriptor.location as is if it is peer scheme. If not, update the location with the final location in this method. You can still return void.

Reply:
done. thanks.